kratos/sources/linear_solver_factory.cpp

```diff
@@ -0,0 +1,123 @@
+        
+        typedef ScalingSolver<SpaceType,  LocalSpaceType> ScalingSolverType;
+       
+        typedef LinearSolverFactoryBase<SpaceType,  LocalSpaceType> LinearSolverFactoryBaseType;
+        typedef LinearSolverFactoryBase<ComplexSpaceType,  ComplexLocalSpaceType> ComplexLinearSolverFactoryBaseType;
+        
+        //NOTE: here we must create persisting objects for the linear solvers
+        static auto CGSolverFactory = LinearSolverFactory<SpaceType,LocalSpaceType,CGSolverType>();
+        static auto BICGSTABSolverFactory = LinearSolverFactory<SpaceType,LocalSpaceType,BICGSTABSolverType>();
+        static auto DeflatedCGSolverFactory= LinearSolverFactory<SpaceType,LocalSpaceType,DeflatedCGSolverType>();
+        static auto SkylineLUFactorizationSolverFactory= LinearSolverFactory<SpaceType,LocalSpaceType,SkylineLUFactorizationSolverType>();
+//         static auto MixedUPLinearSolverFactory = LinearSolverFactory<SpaceType,LocalSpaceType,MixedUPLinearSolverType>();
+        static auto TFQMRSolverFactory = LinearSolverFactory<SpaceType,LocalSpaceType,TFQMRSolverType>();
+        static auto AMGCLSolverFactory= LinearSolverFactory<SpaceType,LocalSpaceType,AMGCLSolverType>();
+        static auto AMGCL_NS_SolverFactory= LinearSolverFactory<SpaceType,LocalSpaceType,AMGCL_NS_SolverType>();
+        static auto ScalingSolverFactory= LinearSolverFactory<SpaceType,LocalSpaceType,ScalingSolverType>();
+        static auto SkylineLUComplexSolverFactory = LinearSolverFactory<ComplexSpaceType, ComplexLocalSpaceType, SkylineLUComplexSolverType>();
+
+        //registration of linear solvers
+//         KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("LinearSolver"), LinearSolverFactory<SpaceType,LocalSpaceType,LinearSolverType>());
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("CGSolver"), CGSolverFactory);
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("BICGSTABSolver"), BICGSTABSolverFactory);
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("DeflatedCGSolver"), DeflatedCGSolverFactory);
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("TFQMRSolver"), TFQMRSolverFactory);
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("SkylineLUFactorizationSolver"),SkylineLUFactorizationSolverFactory );
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("AMGCL"), AMGCLSolverFactory);
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("AMGCLSolver"), AMGCLSolverFactory); //registered with two different names
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("AMGCL_NS_Solver"),AMGCL_NS_SolverFactory );
+        KratosComponents<LinearSolverFactoryBaseType>::Add(std::string("ScalingSolver"),ScalingSolverFactory );
+        KratosComponents<ComplexLinearSolverFactoryBaseType>::Add(std::string("SkylineLUComplexSolver"), SkylineLUComplexSolverFactory);
```

Comment:
@pooyan-dadvand ,maybe we should call this 

       SkylineLUComplexSolverFunctor

instead of 

       SkylineLUComplexSolverFactory